60replicas.py

```diff
@@ -0,0 +1,191 @@
+            try:
+                vm = api.get_vm(vm_name)
+            except:
+                continue
+        vm_list.append(vm)
+    return vm_list
+
+
+def start_vms(hosts):
+    hosts_done = []
+    for host in hosts:
+        hosts_done.append(api.start(host.name, host))
+    return hosts_done
+
+
+def delete_vms(hosts):
+    for host in hosts:
+        thread = Thread(target=api.remove_vm(host.name))
+        thread.start()
+
+
+def stop_vms(hosts):
+    for host in hosts:
+        host.stop()
+
+
+def prepare_inventory(hostlist, inventory_file, config_file, testrc_file):
+    result = []
+    for host in hostlist:
+        result.append({'fqdn': host.fqdn, 'ip': host.ip})
+    with open(inventory_file, 'w') as invfile:
+        hostnames = []
+        for host in result:
+            if host['fqdn'] != '.':
+                hostnames.append(host['fqdn'][:-1])
+        restext = '\n'.join(hostnames)
+        invfile.write(restext)
+
+    domainname = result[0]['fqdn'][:-1].replace('vm', 'dom')
+    ipadomain = ipadomainskel % domainname
+    cleanresult = []
+    for host in result:
+        if host['fqdn'] and host['ip']:
+            cleanresult.append(host)
+    hosts = masterskel % (cleanresult[0]['fqdn'],
+                          cleanresult[0]['fqdn'][:-1],
+                          cleanresult[0]['ip'])
+    testrc_hosts = [cleanresult[0]['fqdn'][:-1]]
+    for host in cleanresult[1:]:
+        hosts += replicaskel % (host['fqdn'],
+                                host['fqdn'][:-1],
+                                host['ip'])
+        testrc_hosts.append(host['fqdn'][:-1])
+
+    with open(config_file, 'w') as configfile:
+        resulttext = header + ipadomain + hosts
+        configfile.write(resulttext)
+    with open(testrc_file, 'w') as testrcfile:
+        testrcfile.write(testrcskel % (testrc_hosts[0],
+                                       " ".join(testrc_hosts[1:]),
+                                       config_file))
+
+
+if __name__ == '__main__':
+    prefix = options.prefix or os.environ['USER']
+    num_replicas = options.num_vms or locals.NUM_VMS
+    vm_suffix = options.suffix or locals.VM_SUFFIX
+    locals.set_locale(options.lab)
+
+    header = "admin_name: admin\n\
+admin_password: %s\n\
+debug: true\n\
+dirman_dn: cn=Directory Manager\n\
+dirman_password: %s\n\
+dns_forwarder: %s\n\
+domain_level: 1\n\
+root_ssh_key_filename: %s\n\
+test_dir: %s\n\
+domains:\n" % (locals.ADMIN_PASSWORD,
+               locals.DIRMAN_PASSWORD,
+               locals.DNS_FORWARDER,
+               locals.PRIVATE_KEY,
+               locals.TEST_DIR)
+
+    ipadomainskel = "- name: %s\n\
+  type: IPA\n\
+  hosts:\n"
+
+    masterskel = "  - name: %s\n\
+    external_hostname: %s\n\
+    ip: %s\n\
+    role: master\n"
+
+    replicaskel = "  - name: %s\n\
+    external_hostname: %s\n\
+    ip: %s\n\
+    role: replica\n"
+
+    testrcskel = "export MASTER=%s\n\
+export REPLICA='%s'\n\
+export IPATEST_YAML_CONFIG=%s"
+
```

Comment:
We can turn these string constants into module level constants. Additionally, use """...""" instead of "\n\..\n\..".